pkg/apis/crd/v1beta1/types.go

```diff
@@ -262,6 +263,95 @@ type IPPoolUsage struct {
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// IPPool defines one or multiple IP sets that can be used for flexible IPAM feature. For instance, the IPs can be
+// allocated to Pods according to IP pool specified in Deployment annotation.
+type IPPool struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the IPPool.
+	Spec IPPoolSpec `json:"spec"`
+
+	// Most recently observed status of the pool.
+	Status IPPoolStatus `json:"status"`
+}
+
+type IPVersion int
+
+const (
+	IPv4 = IPVersion(4)
+	IPv6 = IPVersion(6)
+)
+
+type IPPoolSpec struct {
+	// IP Version for this IP pool - either 4 or 6
+	IPVersion IPVersion `json:"ipVersion"`
```

Comment:
I think @tnqn was wondering if this was still useful?

Reply:
Yes, I wonder if this field really helps anything. It requires users to fill one more field which needs to be consistent with other fields, and requires antrea to do one more validation.
cc @gran-vmv @jianjuns

Reply:
Done